mbus/nats_handler.go

```diff
@@ -130,17 +130,35 @@ func (h *natsHandler) Start(handlerFunc boshhandler.Func) error {
 	h.logger.Info(h.logTag, "Attempting to connect to NATS")
 
 	if net.ParseIP(connectionInfo.IP) != nil {
-		err = h.platform.DeleteARPEntryWithIP(connectionInfo.IP)
-		if err != nil {
-			h.logger.Error(h.logTag, "Cleaning ip-mac address cache for: %s", connectionInfo.IP)
-		}
+		h.arpClean()
 	}
-
 	var natsOptions = []nats.Option{
-		nats.Timeout(natsConnectionTimeout),
+		nats.DisconnectHandler(func(c *nats.Conn) {
+			for c.IsReconnecting() {
+				h.arpClean()
+				h.logger.Debug(natsHandlerLogTag, fmt.Sprintf("Waiting to reconnect to nats.. Connected: %v", c.IsConnected()))
+				time.Sleep(time.Second)
+			}
+			h.logger.Debug(natsHandlerLogTag, "Reconnected to nats")
+		}),
+		nats.ClosedHandler(func(c *nats.Conn) {
```

Comment:
My understanding is that `ClosedHandler` is called when the NATs client failed to create a connection or to reconnect. Is the reconnection use case possible here?

Reply:
@nouseforaname any comments on this?

Reply:
yes, you have a valid point there. I updated the code to only log the last error here. I updated the code to match the nats example from:
https://github.com/nats-io/nats.go#clustered-usage